patterns/ecr-pull-through-cache/variables.tf

```diff
@@ -1,61 +1,26 @@
-# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-# SPDX-License-Identifier: Apache-2.0
-
-variable "region" {
-  description = "AWS region"
-  type        = string
-  default     = "us-east-1"
-}
-
-# defaults to directory name ${basename(path.cwd)}
-variable "name" {
-  description = "EKS Cluster Name and the VPC name"
-  type        = string
-  default     = ""
-}
-
-variable "cluster_version" {
-  type        = string
-  description = "Kubernetes Version"
-  default     = "1.30"
-}
-
-variable "capacity_type" {
-  type        = string
-  description = "Capacity SPOT or ON_DEMAND"
-  default     = "SPOT"
-}
-
 # defaults to data.aws_caller_identity.current.account_id
 variable "ecr_account_id" {
   type        = string
   description = "ECR repository AWS Account ID"
   default     = ""
 }
 
-# defaults to local.region -> var.region
+# defaults to local.region
 variable "ecr_region" {
   type        = string
   description = "ECR repository AWS Region"
   default     = ""
 }
 
 variable "docker_secret" {
```

Comment:
we should remove all variables except for those where we absolutely need users to supply information. Here, that looks like its just the `docker_secret` variable. We should also remove the defaults - we are requiring them to supply this, and by setting a default it will just fail silently